pkg/kv/kvserver/batcheval/cmd_export.go

```diff
@@ -186,16 +186,34 @@ func evalExport(
 				StripTenantPrefix:  true,
 				StripValueChecksum: true,
 			}
-			summary, resume, fingerprint, err = storage.MVCCExportFingerprint(ctx, cArgs.EvalCtx.ClusterSettings(), reader, opts, destFile)
+			var hasRangeKeys bool
+			summary, resume, fingerprint, hasRangeKeys, err = storage.MVCCExportFingerprint(ctx,
+				cArgs.EvalCtx.ClusterSettings(), reader, opts, destFile)
+			if err != nil {
+				if errors.HasType(err, (*storage.ExceedMaxSizeError)(nil)) {
+					err = errors.WithHintf(err,
+						"consider increasing cluster setting %q", MaxExportOverageSetting)
+				}
+				return result.Result{}, err
```

Comment:
It's often easy to lose track of the important details in this and related, functions. If your fancied, perhaps we could extract this to be something like:

```suggestion
				return result.Result{}, maybeAnnotateMaxSizeError(err)
```

Reply:
nice idea, done.